src/app/administrator/override/override.component.ts

```diff
@@ -0,0 +1,53 @@
+import { Component, OnInit, OnChanges } from "@angular/core";
+import { OLmsService } from "src/app/common/o/services/oLms.service";
+import { environment } from "src/environments/environment";
+import * as moment from "moment";
+
+@Component({
+  selector: "app-override",
+  templateUrl: "./override.component.html",
+  styleUrls: ["./override.component.css"],
+})
+export class OverrideComponent implements OnInit {
+  overridableLoans: any[];
+
+  constructor(private lmsService: OLmsService) {}
+
+  ngOnInit() {
+    this.readOverridableLoans();
+  }
+
+  readOverridableLoans() {
+    this.lmsService
+      .get(
+        `${environment.adminBackendUrl}${environment.readOverridableLoansUri}`
+      )
+      .subscribe(
+        (result: object[]) => {
+          this.overridableLoans = result;
+          for (const loan of this.overridableLoans)
+            this.lmsService.processLoan(loan);
+        },
+        (error) => {
+          this.overridableLoans = [];
+          // do something with a logger here
+          alert(error);
+        }
+      );
+  }
+
+  doOverride(loan: any) {
+    const dateOut = moment(loan.dateOut).format("YYYY-MM-DDTHH_mm_ss"),
+      overrideUri = `/loans/book/${loan.bookId}/borrower/${loan.cardNo}/branch/${loan.branchId}/dateout/${dateOut}`;
+    this.lmsService
+      .put(`${environment.adminBackendUrl}${overrideUri}`)
+      .subscribe(
+        () => this.readOverridableLoans(),
+        (error) => {
+          // do something with a logger here
+          alert(error);
```

Comment:
When you had the problem of errors only returning [Object object], when you used a debugger did the object not have properties? Would it return any status code or simply [Object object]?

Reply:
I haven't looked at it with a debugger yet. No status code is shown in the error alerts.

Reply:
If you're still having trouble with that, it's probably just because you are trying to print a json object. It will work printing to the console, but you need to be more specific when displaying it in the html

Reply:
This doesn't display it in HTML; it creates an alert window that pops up on top of the page. However it does seem likely that the problem is as you suggest.